pkg/handler/v2/external_cluster/gke.go

```diff
@@ -1025,12 +1136,13 @@ func getSAFromPreset(ctx context.Context,
 	userInfoGetter provider.UserInfoGetter,
 	presetProvider provider.PresetProvider,
 	presetName string,
+	projectID string,
 ) (string, error) {
 	userInfo, err := userInfoGetter(ctx, "")
```

Comment:
use the projectID here as well, its needed to get the proper roles for the project. 

Although we dont use impersonation client (which uses the roles) to get the presets, better to add it then later look for bugs if the getting Preset implementation changes